### PR TITLE
Fix Faulty Regex pattern in custom DNS domain validation

### LINKED ISSUE #5290: Unable to add custom domains randomly - 5.17 checkDomain regex is broken
### Versions

- Pi-hole:  v5.17
- AdminLTE:  v5.20.1
- FTL:  v5.23

### Platform

- OS and version:  Ubuntu 22 LTS

- Platform:  Proxmox LXC

### Expected behavior

When adding a new custom DNS entry from web GUI, it should add it to the pihole custom list.

### Actual behavior / bug

It says it added it, but it actually doesn't. I traced it to the recent commit https://github.com/pi-hole/pi-hole/commit/b8c3f6d999b3fec6375e09ecbf3430e258ad6294 to checkDomain function.

The change made the domain validation inconsistent. Old regex:
```shell
root@Pi-Hole:~# grep -P "^((-|_)*[a-z\\d]((-|_)*[a-z\\d])*(-|_)*)(\\.(-|_)*([a-z\\d]((-|_)*[a-z\\d])*))*$" <<< "db.rg-10.hm"
db.rg-10.hm
root@Pi-Hole:~# grep -P "^((-|_)*[a-z\\d]((-|_)*[a-z\\d])*(-|_)*)(\\.(-|_)*([a-z\\d]((-|_)*[a-z\\d])*))*$" <<< "my.rg-10.hm"
my.rg-10.hm
root@Pi-Hole:~# grep -P "^((-|_)*[a-z\\d]((-|_)*[a-z\\d])*(-|_)*)(\\.(-|_)*([a-z\\d]((-|_)*[a-z\\d])*))*$" <<< "mydb.rg-10.hm"
mydb.rg-10.hm
```
New regex:
```shell
root@Pi-Hole:~# grep -P "^((-|_)*[a-z0-9]((-|_)*[a-z0-9)*(-|_)*)(\\.(-|_)*([a-z0-9]((-|_)*[a-z0-9])*))*$" <<< "db.rg-10.hm"
root@Pi-Hole:~# grep -P "^((-|_)*[a-z0-9]((-|_)*[a-z0-9)*(-|_)*)(\\.(-|_)*([a-z0-9]((-|_)*[a-z0-9])*))*$" <<< "my.rg-10.hm"
root@Pi-Hole:~# grep -P "^((-|_)*[a-z0-9]((-|_)*[a-z0-9)*(-|_)*)(\\.(-|_)*([a-z0-9]((-|_)*[a-z0-9])*))*$" <<< "mydb.rg-10.hm"
mydb.rg-10.hm
```

#### Issue 1

The regex validation is inconsistent. Or maybe there is a certain rule to follow? If there, some explanation would be great!

#### Issue 2

The web GUI doesn't reflect what `pihole -a addcustomdns` says:

```shell
root@Pi-Hole:~# pihole -a addcustomdns 192.168.1.24 my.rg-10.hm
  [✓] Adding custom DNS entry...
  [✗] Invalid Domain passed!
```

image
image

### Steps to reproduce

1. Go to Local DNS/DNS Records
2. Enter domain and address (you may want to specifically try the above domain, as the new regex is inconsistent)
3. See success banner without an entry in the table below

1. Go to pihole cli
2. Run `pihole -a addcustomdns <ip> <domain>`,  (again, you may want to specifically try the above domain)
3. See that the domain is invalid as per checkDomain, though it was valid <5.17

## Debug Token

- URL:  https://tricorder.pi-hole.net/pKQYemfa/

### Screenshots

Add two above; one for the notification and other for the DBS records.

### Additional context

Updated from 5. something to 5.17 and all my local dns broke. Reinstalled all the pi-hole instances and traced it down to this.

### PR DESCRIPTION
**Pull request by PromoFaux**
### **What does this PR aim to accomplish?:**

Fixes #5290  by adding the `]` back in where it was accidentally removed

```
root@docker-pihole-pi4:/# grep -P "^((-|_)*[a-z0-9]((-|_)*[a-z0-9])*(-|_)*)(\\.(-|_)*([a-z0-9]((-|_)*[a-z0-9])*))*$" <<< "db.rg-10.hm"
db.rg-10.hm
root@docker-pihole-pi4:/# grep -P "^((-|_)*[a-z0-9]((-|_)*[a-z0-9])*(-|_)*)(\\.(-|_)*([a-z0-9]((-|_)*[a-z0-9])*))*$" <<< "my.rg-10.hm"
my.rg-10.hm
root@docker-pihole-pi4:/# grep -P "^((-|_)*[a-z0-9]((-|_)*[a-z0-9])*(-|_)*)(\\.(-|_)*([a-z0-9]((-|_)*[a-z0-9])*))*$" <<< "mydb.rg-10.hm"
mydb.rg-10.hm
```

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

File: test/test_any_automated_install.py [hunk 1]
```diff
@@ -1151,3 +1151,15 @@
...
+def test_webpage_sh_valid_domain(host):
+    """Confirms webpage.sh returns 0 for a valid domain"""
+    output = host.run(
+        """
+    source /opt/pihole/webpage.sh
+    checkDomain "pi-hole.net"
+    """
+    )
+
+    assert "pi-hole.net" in output.stdout
```
[the remark this answers is not in the document]
Comment (PromoFaux) [reply test/test_any_automated_install.py]: No.

File: test/test_any_automated_install.py [hunk 2]
```diff
@@ -1151,3 +1151,30 @@
...
+    check3 = host.run(
+        """
+    source /opt/pihole/webpage.sh
+    checkDomain "abc.pi-hole.net"
+    """
+    )
+
+    assert "pi-hole.net" in check1.stdout
+    assert "ab.pi-hole.net" in check2.stdout
+    assert "abc.pi-hole.net" in check3.stdout
```
[the remark this answers is not in the document]
Comment (rdwebdesign) [reply test/test_any_automated_install.py:1180]: Hmmmmmm...

**Review by rdwebdesign — APPROVED** [review]

**Comment by yubiuser** [comment]
Do we ignore the code scanning errors? They seem to be false positives.

**Comment by PromoFaux** [comment]
Yeah, I already dismissed them as false positive/test values, though they stayed on the screen.

**Comment by Gopikrishna19** [comment]
@PromoFaux thank you for taking care of it! I wish I had known how to make this change.

**Pull request merged**